test/threads_exec.jl

```diff
@@ -722,15 +722,65 @@ let a = zeros(nthreads())
 end
 
 # static schedule
-function _atthreads_static_schedule()
+function _atthreads_static_schedule(n)
+    ids = zeros(Int, n)
+    Threads.@threads :static for i = 1:n
+        ids[i] = Threads.threadid()
+    end
+    return ids
+end
+@test _atthreads_static_schedule(nthreads()) == [1:nthreads();]
+@test _atthreads_static_schedule(1) == [1;]
+@test_throws TaskFailedException @threads for i = 1:1; _atthreads_static_schedule(nthreads()); end
```

Comment:
I know `@test_throws TaskFailedException ...` is what was previously used. But, while we are at it, how about testing the message which is more robust?

```suggestion
@test_throws(
    "`@threads :static` cannot be used concurrently or nested",
    @threads for i = 1:1; _atthreads_static_schedule(nthreads()); end,
)
```